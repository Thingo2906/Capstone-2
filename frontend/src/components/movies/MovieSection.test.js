import React from "react";
import { render } from "@testing-library/react";
import MovieSection from "./MovieSection";
import MovieApi from "../../api/api";




describe("MovieSection Component", () => {
    // We can change the fetchData for different category of movie
  it("renders without crashing", () => {
    render(<MovieSection fetchData={MovieApi.getUpcoming} />);
  });
  it("renders the title", () => {
     const { asFragment, getByText } = render(<MovieSection title="Upcoming" fetchData={MovieApi.getUpcoming} />);
     expect(asFragment()).toMatchSnapshot();
     const titleElement = getByText("Upcoming");
     expect(titleElement).toBeInTheDocument();
   });


})



