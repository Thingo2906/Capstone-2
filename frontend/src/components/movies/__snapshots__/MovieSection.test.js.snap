// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSection Component renders the title 1`] = `
<DocumentFragment>
  <div
    class="movieList"
  >
    <h2>
      Upcoming
    </h2>
    <div
      class="poster-list"
    />
  </div>
</DocumentFragment>
`;
